e2e/packages/contracts/src/systems/CustomStructEnumSystem.sol

```diff
@@ -14,11 +14,8 @@ contract CustomStructEnumSystem is System {
     VAL2
   }
 
-  TestStruct internal testStruct;
-  TestEnum internal testEnum;
-
-  function stub3(uint8 val1, bytes32 val2) public {
-    testStruct = TestStruct({ val1: val1, val2: val2 });
-    testEnum = TestEnum.VAL2;
-  }
+  // TODO: this fails to codegen an interface
+  // function stub3(TestStruct memory testStruct, TestEnum testEnum) public {
```

Comment:
Running codegen with this line throws this error:
```
Function "stub3" in contract "CustomStructEnumSystem": Symbol "TestStruct" has no explicit import
```

We require things like structs to be defined outside of systems and imported, to ensure they're the same object/type. Defining them in multiple places means different types. And we have no way in the current codegen pipeline for a system to also inherit from the generated interface (and it would be awkward anyway, since we'd overwrite the file you're meant to be editing).